projects/hydro-protocol/index.js

```diff
@@ -1,10 +1,65 @@
-const { queryContract } = require("../helper/chain/cosmos");
+const { queryContract, totalSupply } = require("../helper/chain/cosmos");
 const ADDRESSES = require("../helper/coreAssets.json");
+const { getClient } = require("../helper/chain/injective");
+const { IndexerGrpcMitoApi, spotPriceFromChainPrice, spotQuantityFromChainQuantity, IndexerGrpcSpotApi } = require('@injectivelabs/sdk-ts')
+const BigNumber = require("bignumber.js");
+const { get } = require("../helper/http");
+
+const hinj = "inj18luqttqyckgpddndh8hvaq25d5nfwjc78m56lc"
+const autoCompound = "inj1mjcg8a73904rj4w7t5qkgn0apua98n059nufma"
+const hdro = "factory/inj1etz0laas6h7vemg3qtd67jpr6lh8v7xz7gfzqw/hdro"
+const xhdro = "inj1qc2tw477wwuvkad0h3g78xqgwx4k8knat6vz0h"
+const hdroInjMarket = "0xc8fafa1fcab27e16da20e98b4dc9dda45320418c27db80663b21edac72f3b597"
+
+const injectiveSpotApi = new IndexerGrpcSpotApi("https://sentry.exchange.grpc-web.injective.network:443")
+const injectiveClient = getClient()
 
-const autocompoundContract = 'inj1mjcg8a73904rj4w7t5qkgn0apua98n059nufma'
 async function tvl(_, _1, _2, { api }) {
-  const { total_supply } = await queryContract({ chain: api.chain, contract: "inj18luqttqyckgpddndh8hvaq25d5nfwjc78m56lc", data: { token_info: {} } })
-  api.add(ADDRESSES.injective.INJ, total_supply)
+  const { total_supply } = await queryContract({ chain: api.chain, contract: hinj, data: { token_info: {} } })
+  const { total_bonded } = await queryContract({ chain: api.chain, contract: autoCompound, data: { state: {} } })
```

Comment:
why are you including the INJ in autocompounder, arent they converted to hINJ, so already counted as part of hINJ supply/tvl?

Reply:
No Auto-compound does not use hINJ. It uses INJ.
hINJ is used in LSD & Farm contract only

Reply:
Ah, I am confused now, what does it do with the inj staked there/how does it compound?

Reply:
Auto-compound contract restake its inj every day.

Reply:
it restakes to hINJ ?

Reply:
No

Reply:
Auto-compound does not use hINJ

Reply:
It restakes using a different staking contract?

Reply:
No auto-compound contract restake itself

Reply:
Hydro-protocol provide two different type INJ staking.
First, LSD (hINJ) type that you know. If user stake INJ to LSD contract then it gives hINJ to user.
Second, Auto-compound (INJ) type. If user stake INJ to Auto-compound contract then it gives nothing to user. Auto-compound contract stake users' INJ and execute (claim inj reward) + (stake) messages every day.

Reply:
thanks for patiently explaining it